src/providers/email.js

```diff
@@ -22,28 +22,25 @@ export default (options) => {
   }
 }
 
-const sendVerificationRequest = ({ identifier: email, url, baseUrl, provider }) => {
-  return new Promise((resolve, reject) => {
-    const { server, from } = provider
-    // Strip protocol from URL and use domain as site name
-    const site = baseUrl.replace(/^https?:\/\//, '')
-
-    nodemailer
+async function sendVerificationRequest ({ identifier: email, url, baseUrl, provider }) {
+  const { server, from } = provider
+  // Strip protocol from URL and use domain as site name
+  const site = baseUrl.replace(/^https?:\/\//, '')
+  try {
+    const nodemailer = requireOptional('nodemailer')
+    await nodemailer
```

Comment:
why not with `const` functions?

```suggestion
const sendVerificationRequest = async ({ identifier: email, url, baseUrl, provider }) => {
  const { server, from } = provider
  // Strip protocol from URL and use domain as site name
  const site = baseUrl?.replace(/^https?:\/\//, '')
  try {
    const nodemailer = requireOptional('nodemailer')
    await nodemailer
```